src/jetpack-editor-setup.js

```diff
@@ -177,8 +177,7 @@ const setupStringsOverrides = () => {
 
 			if (
 				onlyCoreBlocks &&
-				jetpackBlockNames.includes( blockName ) &&
-				videoPressBlock
+				( jetpackBlockNames.includes( blockName ) || videoPressBlock )
```

Comment:
This condition checks if the block passed is a Jetpack block, so it needed to be fixed to meet the expectation. Seems we overlooked this bug when introducing these changes (https://github.com/wordpress-mobile/gutenberg-mobile/pull/5459/files#diff-b38d335848f5255ead813ac6b271ead0d6afc53cca251ade416d02903c367b05) 🤦 .

Reply:
> Is there a way for us to better structure this logic to reduce the need for this one-off check?

Good point. We can definitely improve this logic and simplify it.

> My understanding is that we cannot include the `jetpack/` prefix in `supportedJetpackBlocks` because [`getJetpackExtensionAvailability`](https://github.com/Automattic/jetpack/blob/58cc1a05fd388c7ddc3cf3936d1db1eaf9079c73/projects/js-packages/shared-extension-utils/src/get-jetpack-extension-availability.js#L6) requires its absence.

Yes, that's correct. We'd need to maintain the format of block names in `supportedJetpackBlocks` as-is. These values are used as the `available_blocks` parameter of Jetpack's initial state, which is being used in helpers like [`getJetpackExtensionAvailability`](https://github.com/Automattic/jetpack/blob/58cc1a05fd388c7ddc3cf3936d1db1eaf9079c73/projects/js-packages/shared-extension-utils/src/get-jetpack-extension-availability.js#L6):

https://github.com/wordpress-mobile/gutenberg-mobile/blob/8cd476d1d8d151cf058b6f8067ef7cc28594367a/src/jetpack-editor-setup.js#L47-L49

> **Idea A:** We might conditionally add the `jetpack/` prefix when mapping if the block name fails a test for a prefix.

That's a good idea. Following the code you suggested, we could check the block's name format and include the `jetpack` extension only for those that don't contain the `/` character. The below suggestion is based on the one shared https://github.com/wordpress-mobile/gutenberg-mobile/pull/6269#discussion_r1350277777 but the regular expression only tests for no-slash block names:

```js
const jetpackBlockNames = Object.keys( supportedJetpackBlocks ).map(
( name ) =>
	// Add `jetpack` prefix if the block name doesn't contain a prefix.
	/^(\w|-)+$/.test( name ) ? `jetpack/${ name }` : name
);
```

> **Idea B:** We go the opposite direction and strip all block prefixes when setting `jetpackEditorInitialState`.

This would be another option. I don't have a strong preference but I'd lean toward **Idea A** to keep `supportedJetpackBlocks` as-is.

Reply:
I applied the above suggestion in https://github.com/wordpress-mobile/gutenberg-mobile/pull/6269/commits/4b17ffbe1cdc6503dd7611b84741911abd4f1d50.

Reply:
Your alternative regex for **Idea A** looks good to me. 👍🏻 

Maybe rename `jetpackBlockNames` to `namespacedBlockNames` to communicate the mapping result? That might even mitigate the need for an inline comment.

Reply:
Good idea, I've just applied this suggestion in https://github.com/wordpress-mobile/gutenberg-mobile/pull/6269/commits/530876a09d46be9c356cca9be4c27764863bd5e9.